### PR TITLE
Ignores tgui button clicks after an update has been triggered on the ui but not sent yet

### PR DESCRIPTION
**Pull request by Cyberboss**
This prevents a case where you could call the ui_act for a disabled button before it actually became disabled.

Fixes the beaker runtimes

Closes #31443

### CODE REVIEW

File: tgui/src/components/button.ract [hunk 1]
```diff
@@ -40,6 +40,7 @@
     oninit () {
       this.on('press', (event) => {
         const { action, params } = this.get()
+        params.nounce = this.get('config.current_nounce')
```
Comment (Cyberboss) [inline tgui/src/components/button.ract]: I can't figure out how to add a field to params here, send help

**Comment by kevinz000** [comment]
Isn't this going to cause legitimate dropped clicks every 0.9 seconds even if nothing changed?

**Comment by Cyberboss** [comment]
Fixed

**Comment by MrStonedOne** [comment]
this isn't quite a solution. if the button being disabled isn't being checked server side, you've already fucked up.

And with canisters this would break alot of shit since they can update every second with pressure changes, and it would likely cause a bunch of accidental plasma flooding in toxins from when the close valve button gets clicked but ignored.

3/10

**Pull request closed without merging**